### PR TITLE
Add documentation for using Winbind as a backend

### PR DESCRIPTION
**Pull request by GabrielNagy**
DEENG-493

### CODE REVIEW

File: conf.example/adsys.yaml [hunk 1]
```diff
@@ -20,5 +20,10 @@ sssd:
   config: /etc/sssd.conf
   cache_dir: /var/lib/sss/db
 
+# Winbind configuration
```
Comment (didrocks) [inline conf.example/adsys.yaml:23]: Maybe add a comment like "if backend: winbind is selected"?

File: doc/02.-Prerequisites.md [hunk 2]
```diff
@@ -35,14 +35,31 @@ At the end of the installation you can reboot the machine and you are ready to l
 
 If anything goes wrong with the join process during installation, you will be notified by a dialog box. You can still reboot the machine, log in as the administrator user of the machine (ie the user you entered in the page **"Who are you?"**) and troubleshoot the issue. The [Ubuntu Server Guide](https://ubuntu.com/server/docs/service-sssd) provides instructions to perform such troubleshooting.
 
-### Joining manually
+### Joining manually using SSSD
 
 The purpose of this document is to describe how to operate ADSys. So we won’t do an in depth description of the operations to manually configure a connection to Active Directory from an Ubuntu Client.
 
-Authentication of Ubuntu against the Active Directory server requires to configure SSSD and Kerberos. SSSD will then retrieve the credentials and the initial security policy of the `Default Domain Policy` .
```
Comment (didrocks) [inline doc/02.-Prerequisites.md:42]: attention to details :) Thanks!

File: doc/02.-Prerequisites.md [hunk 3]
```diff
@@ -35,14 +35,31 @@ At the end of the installation you can reboot the machine and you are ready to l
...
+
+In addition to SSSD, ADSys supports using Winbind as a backend. The easiest way to join a domain using Winbind is to use the `realmd` utility, as described in the [Samba - Active Directory](https://ubuntu.com/server/docs/samba-active-directory) guide.
+
+ADSys uses SSSD as a default backend, so Winbind has to be opted into explicitly via the following config option in `adsys.yaml`:
+
+```yaml
+ad_backend: winbind
+```
+
+In addition, Winbind requires additional dependencies to be installed. They can be installed by executing the following command, prior to ADSys:
```
Comment (didrocks) [inline doc/02.-Prerequisites.md:56]: add something like, "for instance, on ubuntu…"

File: doc/02.-Prerequisites.md [hunk 4]
```diff
@@ -60,8 +77,22 @@ Reboot then to allow the machine to do its policy refresh.
...
+
+By default, there is no default domain configured in SSSD. You have to enter the full user name with one of the forms `USER@DOMAIN.COM` or `DOMAIN/USER`.
 
 On the first log in the user's home directory is created.
 
 All of this (default domain, default path for home directories, default shell, etc.) is configurable in `/etc/sssd/sssd.conf`.
+
+### Winbind
+
+If Winbind is used as a backend, the account has to be specified in the following format: `DOMAIN\\USER`.
```
Comment (didrocks) [inline doc/02.-Prerequisites.md:90]: hum, winbind doesn’t support USER@DOMAIN too? I don’t remember with the amount of testing we did, you are probably right, but just checking :)

File: doc/10.-The-adsys-daemon.md [hunk 5]
```diff
@@ -117,6 +117,11 @@ sssd:
   config: /etc/sssd.conf
   cache_dir: /var/lib/sss/db
 
+# Winbind configuration
```
Comment (didrocks) [inline doc/10.-The-adsys-daemon.md:120]: check the comment on the config too. (I start to think we should automate this update too)

**Review by didrocks — CHANGES_REQUESTED** [review]
Really good update!

* one (+ its repetition) or a real comment
* one precision that we want to specify ubuntu specific instructions
* one question to double check
* one random "thank" comment :)

File: doc/02.-Prerequisites.md (hunk 4 above)
Comment (didrocks) [reply doc/02.-Prerequisites.md]: \o/

**Review by didrocks — APPROVED** [review]
Excellent job!

**Review by denisonbarbosa — APPROVED** [review]
LGTM. Well done!

**Pull request merged**